eng/build.sh

```diff
@@ -570,4 +581,19 @@ export DOTNETSDK_ALLOW_TARGETING_PACK_CACHING=0
 cmakeargs="${cmakeargs// /%20}"
 arguments="$arguments /p:TargetArchitecture=$arch /p:BuildArchitecture=$hostArch"
 arguments="$arguments /p:CMakeArgs=\"$cmakeargs\" $extraargs"
+
+if [[ "$bootstrap" == "1" ]]; then
+  # Strip build actions other than -restore and -build from the arguments for the bootstrap build.
+  bootstrapArguments="${arguments//-sign/}"
+  bootstrapArguments="${bootstrapArguments//-publish/}"
+  bootstrapArguments="${bootstrapArguments//-pack/}"
```

Comment:
To fix MSB1001 `Switch: -`, update line 312 to use `-pack` instead of `--pack` (it's the odd one out given line 550 `-restore` and `-build`).

Reply:
@jkoritzinsky if we are going with stripping double too, then those should happen before single hyphen ones to fix  MSB1001 `Switch: -` error.